core/bindings.rs

```diff
@@ -757,6 +783,52 @@ pub fn module_resolve_callback<'s>(
   None
 }
 
+// Sets the V8 promise hook, allowing tracking of context across
+// Promise chains and async functions
+fn set_promise_hook(
+  scope: &mut v8::HandleScope,
+  args: v8::FunctionCallbackArguments,
+  _rv: v8::ReturnValue,
+) {
+  let hook = match v8::Local::<v8::Function>::try_from(args.get(0)) {
+    Ok(val) => val,
+    Err(_) => {
+      let msg = v8::String::new(scope, "Invalid argument").unwrap();
+      let exception = v8::Exception::type_error(scope, msg);
+      scope.throw_exception(exception);
+      return;
```

Comment:
For a future PR: consider moving this into a helper function. I count 5 or 6 other instances of the exact same pattern.